### PR TITLE
First attempt at a schedule function

### PR DESCRIPTION
**Pull request by phamann**
This creates a scheduler function which will be triggered once an hour and uses `moment.js` timezone support to ascertain whether the current hour maps to a given time in certain timezones and invokes a given function is true. 

We've had to do this as the default AWS Lambda scheduling only supports UTC and therefore doesn't account for daylight saving.

- Model times in `functions/scheduler/lib/schedule.js`
- Add logic to `functions/scheduler/index.js`

#### N.b.
Tests to follow :wink: need to prove fetching the `functionName` from the context works first.

### CODE REVIEW

File: functions/scheduler/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+function format(data) {
+	return Object.assign(data[1], { zone: data[0] });
+}
+
+function isCurrentHour(data) {
+	return moment().tz(data.zone).format('HH') === data.time;
+}
+
+function invoke(prefix) {
+	console.log('PREFIX', prefix);
```
Comment (laurieboyes) [inline functions/scheduler/index.js]: Should we be using `require('@financial-times/n-lambda').logger` instead of `console.log`?

Comment (matthew-andrews) [inline functions/scheduler/index.js]: yup^^

**Comment by laurieboyes** [comment]
👍

File: functions/scheduler/lib/lambda.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+const AWS = require('aws-sdk');
+const denodeify = require('denodeify');
+
+const lambda = new AWS.Lambda({region: 'eu-west-1'});
+const invoke = denodeify(lambda.invoke.bind(lambda));
+
+module.exports = {
+	invoke: data => invoke(data)
```
Comment (keirog) [inline functions/scheduler/lib/lambda.js]: `{ invoke }`?

**Pull request merged**